### PR TITLE
Declare only 1 source for logical table source on smsusage dataset

### PR DESCRIPTION
**Pull request by jimleroyer**
# Summary | Résumé

On my last attempt to fix the TF changes to left join both smsusage and notifications dataset of quicksight, we got the following error:

```
Message_: "Exactly one of PhysicalTableId, JoinInstruction or DataSetArn must be set for LogicalTable"
```

Hence it seems that declaring the `PhysicalTableId` and a `JoinInstruction` are considered two different sources, even though the physical table is actually used in the join. I assume that the join instruction will get the references right. The documentation did not declare this constraint by the way, of having only one of these. Let's try this to see if that will fix our current issue and move on to the next one.

## Related Issues | Cartes liées

* https://app.zenhub.com/workspaces/notify-planning-core-6411dfb7c95fb80014e0cab0/issues/gh/cds-snc/notification-planning-core/309

# Test instructions | Instructions pour tester la modification

* Plan should pass.
* The terragrunt apply op should pass.

# Release Instructions | Instructions pour le déploiement

Monitor the `merge to main` github action closely to make sure it is successful. 

# Reviewer checklist | Liste de vérification du réviseur

- [ ] This PR does not break existing functionality.
- [ ] This PR does not violate GCNotify's privacy policies.
- [ ] This PR does not raise new security concerns. Refer to our GC Notify Risk Register document on our Google drive.
- [ ] This PR does not significantly alter performance.
- [ ] Additional required documentation resulting of these changes is covered (such as the README, setup instructions, a related ADR or the technical documentation).

> ⚠ If boxes cannot be checked off before merging the PR, they should be moved to the "Release Instructions" section with appropriate steps required to verify before release. For example, changes to celery code may require tests on staging to verify that performance has not been affected.

### CODE REVIEW

File: aws/quicksight/dataset_sms_usage.tf [hunk 1]
```diff
@@ -54,7 +54,12 @@ resource "aws_quicksight_data_set" "sms_usage" {
     alias = "smsusage"
 
     source {
-      physical_table_id = "smsusage"
```
Comment (jimleroyer) [inline aws/quicksight/dataset_sms_usage.tf:57]: The error message that both physical table id and join instructions were present when only one should be. So we are removing that one!

**Review by ben851 — APPROVED** [review]
WORK!

File: aws/quicksight/dataset_sms_usage.tf (hunk 1 above)
Comment (sastels) [reply aws/quicksight/dataset_sms_usage.tf:57]: riiiight.... it wants just one think as the "source", ie here we want the source to be the join, basically.

I thought there also needed to be some code around what fields would appear in the final table... though maybe it's just going to use all of them if you don't restrict it? Best to apply on dev and see what happens 😿

**Review by sastels — APPROVED** [review]

**Comment by jimleroyer** [comment]
Closing this as more recent changes went past the issue with the join by using CloudFormation instead of the Terraform QuickSight module.

**Pull request closed without merging**